azure_arc_servers_jumpstart/docs/aws_terraform_ubuntu.md

```diff
@@ -0,0 +1,146 @@
+    
+    * TF_VAR_subscription_id=Your Azure Subscription ID
+    * TF_VAR_client_id=Your Azure Service Principal app id
+    * TF_VAR_client_secret=Your Azure Service Principal password
+    * TF_VAR_tenant_id=Your Azure tenant ID
+    * AWS_ACCESS_KEY_ID=AWS access key
+    * AWS_SECRET_ACCESS_KEY=AWS secret key
+
+* From CLI, navigate to the [*azure_arc_servers_jumpstart/aws/ubuntu/terraform*](../aws/ubuntu/terraform) directory of the cloned repo.
+
+* Export the environment variables you edited by running [*scripts/vars.sh*](../aws/ubuntu/terraform/scripts/vars.sh) with the source command as shown below. Terraform requires these to be set for the plan to execute properly. Note that this script will also be automatically executed remotely on the AWS virtual machine as part of the Terraform deployment. 
+
+    ```source ./scripts/vars.sh```
+
+* Make sure your SSH keys are available in *~/.ssh* and named *id_rsa.pub* and *id_rsa*. If you followed the ssh-keygen guide above to create your key then this should already be setup correctly. If not, you may need to modify [*main.tf*](../aws/ubuntu/terraform/main.tf) to use a key with a different path.
+
+* Run the ```terraform init``` command which will download the Terraform AzureRM provider.
+
+    ![](../img/aws_ubuntu/09.png)
+
+# Deployment
+
+* Run the ```terraform apply --auto-approve``` command and wait for the plan to finish. Upon completion, you will have an AWS Amazon Linux 2 VM deployed and connected as a new Azure Arc server inside a new Resource Group.
+
+* Open the Azure portal and navigate to the resource group "Arc-Servers-Demo". The virtual machine created in AWS will be visible as a resource.
+
+    ![](../img/aws_ubuntu/10.png)
+
+# Semi-Automated Deployment (Optional)
+
+As you may have noticed, the last step of the run is to register the VM as a new Arc server resource.
+    ![](../img/aws_ubuntu/11.png)
+
+If you want to demo/control the actual registration process, do the following: 
+
+1. In the [*install_arc_agent.sh.tmpl*](../aws/ubuntu/terraform/scripts/install_arc_agent.sh.tmpl) script template, comment out the "Run connect command" section and save the file.
+
+    ![](../img/aws_ubuntu/12.png)
+
+2. Get the public IP of the AWS VM by running ```terraform output```
+
+    ![](../img/aws_ubuntu/13.png)
+
+3. SSH the VM using the ```ssh ubuntu@x.x.x.x``` where x.x.x.x is the host ip. 
+
+    ![](../img/aws_ubuntu/14.png)
+
+4. Export all the environment variables in [*vars.sh*](../aws/ubuntu/terraform/scripts/vars.sh)
+
+    ![](../img/aws_ubuntu/15.png)
+
+5. Run the following command
+    ```azcmagent connect --service-principal-id $TF_VAR_client_id --service-principal-secret $TF_VAR_client_secret --resource-group "Arc-Servers-Demo" --tenant-id $TF_VAR_tenant_id --location "westus2" --subscription-id $TF_VAR_subscription_id```
```

Comment:
```bash
azcmagent connect --service-principal-id $TF_VAR_client_id --service-principal-secret $TF_VAR_client_secret --resource-group "Arc-Servers-Demo" --tenant-id $TF_VAR_tenant_id --location "westus2" --subscription-id $TF_VAR_subscription_id
```

Reply:
not sure what you are asking for in this comment. Can you elaborate?